mlflow/pyfunc/__init__.py

```diff
@@ -799,9 +800,22 @@ def _predict(self, data: PyFuncInput, params: Optional[dict[str, Any]] = None) -
         # fetch the schema from metadata to avoid signature change after model is loaded
         self.input_schema = self.metadata.get_input_schema()
         self.params_schema = self.metadata.get_params_schema()
-        data, params = _validate_prediction_input(
-            data, params, self.input_schema, self.params_schema, self.loader_module
-        )
+        # signature can only be inferred from type hints if the model is PythonModel
+        if self.metadata._is_signature_from_type_hint():
+            from mlflow.types.type_hints import (
```

Comment:
q: why do we need to import these modules inside the method?

Reply:
To avoid breaking mlflow skinny where numpy is not a dependency